### ISSUE TITLE
Unable to Register Runner

### ISSUE THREAD

**Issue by alacava** [Chocobozzz/PeerTube#6171]
### Describe the current behavior

When I try to register a runner I get:
Cannot register this PeerTube runner.
Error: This runner is not currently running in server mode on this system. Please run it using the `server` command first (in another terminal for example) and then retry your command.
    at IPCClient.run (file:///usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:74058:15)
    at async registerRunner (file:///usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:74826:3)
    at async _Command.<anonymous> (file:///usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:89272:5)

On the peertube-runner server app side, I get:
peertube-runner server
[17:33:02.082] INFO (4730): Running PeerTube runner in server mode
[17:33:02.086] INFO (4730): Using /root/.config/peertube-runner-nodejs/default/config.toml as configuration file
[17:33:02.094] INFO (4730): IPC socket created on /root/.local/share/peertube-runner-nodejs/default/peertube-runner.sock
[17:33:02.095] INFO (4730): Using /root/.cache/peertube-runner-nodejs/default/transcoding for transcoding directory
[17:33:23.067] ERROR (4730): uncaughtException
    origin: "uncaughtException"
    err: {
      "type": "TypeError",
      "message": "Cannot read properties of undefined (reading 'data')",
      "stack":
          TypeError: Cannot read properties of undefined (reading 'data')
              at Connection._untag (file:///usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:10508:38)
              at Connection._read (file:///usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:10429:32)
              at Connection._init (file:///usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:10895:16)
              at Socket.emit (node:events:518:28)
              at emitReadable_ (node:internal/streams/readable:832:12)
              at process.processTicksAndRejections (node:internal/process/task_queues:81:21)
    }
[17:33:23.078] INFO (4730): Cleaning up after program exit 

### Steps to reproduce

1. Followed docs page
2. Run peertube-runner register --url https://xyz.com --registration-token ptrrt-xxx --runner-name peertube-runner
3. Get error message

I'm seeing this problem on different runners and different peertube servers. 

### Describe the expected behavior

The peertube runner registers. 

### Additional information

* PeerTube instance:
  * URL:
  * Version: 6.0.2
  * NodeJS version: 20.11.0
  * Ffmpeg version: 4.4.2-0ubuntu0.22.04.1

* Browser name, version and platforms on which you could reproduce the bug: NA
* Link to browser console log if relevant: NA
* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`):  See above.

**Comment by ROBERT-MCDOWELL** [comment]
you should read the docs
https://docs.joinpeertube.org/maintain/tools

**Comment by alacava** [comment]
@ROBERT-MCDOWELL I did read the docs. I've had it working before so I know I have the right process down.

**Comment by ROBERT-MCDOWELL** [comment]
so show your config and what commands you did

**Comment by Chocobozzz** [comment]
Hi,

Can you try to run the runner server with another user than `root`?

**Comment by KuJoe** [comment]
I'm in the same boat, attempts to register a remote runner are failing for me.

> # sudo -u prunner peertube-runner register --url https://mywebsite --registration-token ptrrt-<key> --runner-name runner01
Cannot register this PeerTube runner.
unknown error

**Comment by Chocobozzz** [comment]
@KuJoe & @alacava can you display the output of `node -v`

Can you also display the output of `ls /usr/lib/node_modules/ | grep msgpack` and ``ls /usr/lib/node_modules/ | grep zlib`?

**Comment by KuJoe** [comment]
> @KuJoe & @alacava can you display the output of `node -v`?
> 
> Can you also display the output of `ls /usr/lib/node_modules/ | grep msgpack` and `ls /usr/lib/node_modules/ | grep zlib`?

Here you go:

```
run01:~# node -v
v20.11.0
run01:~# ls /usr/lib/node_modules/ | grep msgpack
run01:~# ls /usr/lib/node_modules/ | grep zlib
run01:~#
```

**Comment by alacava** [comment]
> @KuJoe & @alacava can you display the output of `node -v`?
> 
> Can you also display the output of `ls /usr/lib/node_modules/ | grep msgpack` and `ls /usr/lib/node_modules/ | grep zlib`?

Here you go as well....
root@ip-10-10-11-144:~# node -v
v20.11.0
root@ip-10-10-11-144:~# ls /usr/lib/node_modules/ | grep msgpack
root@ip-10-10-11-144:~# ls /usr/lib/node_modules/ | grep zlib
root@ip-10-10-11-144:~#

**Comment by KuJoe** [comment]
I tried it with different versions of NodeJS on clean OS installs since it was something @alacava and I had in common without any luck. I'm running Ubuntu 22.04 but I'm going to give a different distro a try to see if that helps.

**Comment by KuJoe** [comment]
It appears to be something specific to Ubuntu 22.04, I followed the guide using a clean install of AlmaLinux 9.3 and aside from a few minor extra steps it registered successfully.

`run01 ~]$ peertube-runner register --url https://mysite --registration-token ptrrt-... --runner-name runner01
PeerTube instance registered`

**Comment by ROBERT-MCDOWELL** [comment]
did you try to disable selinux?

**Comment by Chocobozzz** [comment]
Can someone give me access to your server so I can debug the issue? You can send me credentials by email (in my github profile)

**Comment by alacava** [comment]
Just emailed you the login for a test instance I setup that you can play with.

**Comment by Chocobozzz** [comment]
Thanks. I was able to debug the ipc code and found the issue is a change in node 20.11 that breaks the ipc library we use.

I [created a PR](https://github.com/timotejroiko/net-ipc/pull/7) on the ipc repository and published a fix available in peertube-runner 0.0.10
